### PR TITLE
executor: migrate test-infra to testify for show_placement_test

### LINKED ISSUE #28610: migrate test-infra to testify for `show_placement_test.go`

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #28610

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: executor/show_placement_test.go [hunk 1]
```diff
@@ -266,19 +271,21 @@ func (s *testSuite5) TestShowPlacementForTableAndPartition(c *C) {
 
 	// not exists
 	err := tk.ExecToErr("show placement for table tn")
-	c.Assert(err.Error(), Equals, "[schema:1146]Table 'test.tn' doesn't exist")
+	require.Equal(t, "[schema:1146]Table 'test.tn' doesn't exist", err.Error())
```
Comment (tisonkun) [inline executor/show_placement_test.go:274]: `require.EqualError`

ditto others.

Comment (hawkingrei) [reply executor/show_placement_test.go]: Thank you. Done.

**Review by tisonkun — APPROVED** [review]

**Review by Defined2014 — APPROVED** [review]

**Pull request merged**